### PR TITLE
Added action-chain click option

### PR DESCRIPTION
**Pull request by JonKoser**
This PR introduces an option to `Click Element` which enables a user to use Selenium ActionChains to issue a click action on an element instead of using the <web_element>.click() function. This enables a more "native" click of any element in the UI. This is in response to issue #1463

### CODE REVIEW

File: src/SeleniumLibrary/keywords/element.py
Comment (aaltat) [inline src/SeleniumLibrary/keywords/element.py]: Needs explanation about order of arguments, which one overrules which one. Now the implementation is done that `action_chain` is applied before `modifier` argument. But I think arguments are applied in wrong order. If `action_chain` and `modifier` are defined with valid values, then `modifier` should be applied and `action_chain` be ignored.

File: src/SeleniumLibrary/keywords/element.py
Comment (aaltat) [inline src/SeleniumLibrary/keywords/element.py]: I would move this logic to an own method.

**Review by aaltat — CHANGES_REQUESTED** [review]
Needs also testing, in addition to comments. Most likely easiest way is to write few acceptance tests and use status checker for verifying correct messages are displayed in the logs.

**Comment by JonKoser** [comment]
Added tests and addressed comments

File: src/SeleniumLibrary/keywords/element.py
Comment (JonKoser) [reply src/SeleniumLibrary/keywords/element.py]: Good call

File: src/SeleniumLibrary/keywords/element.py
Comment (JonKoser) [reply src/SeleniumLibrary/keywords/element.py]: Fix in

**Comment by aaltat** [comment]
The test are good start but they need to look deeper under the hood. Now Test ensures that click happens, but tests do not ensure that click happens by using action chains. Usually in this kind of situation, it is easiest to use [statuschecker](https://github.com/robotframework/statuschecker/) to verify that logs from Selenium are from actions chains and not from normal `click()`. Example [Choose File](https://github.com/robotframework/SeleniumLibrary/blob/master/atest/acceptance/keywords/choose_file.robot#L35) keyword uses it for same purposes.

**Comment by JonKoser** [comment]
Tests guarantee log output now

File: atest/acceptance/keywords/click_element.robot
Comment (aaltat) [inline atest/acceptance/keywords/click_element.robot]: Better but not quite on what I had in my mind. Test are done, among other things, to safe guard maintainers from stupid mistakes. Now I can change the action chains implementation and it's not always certain that test would capture the accidental change. 

The test now ensures that keyword logs by using the Robot Framework API (which is relevant too) and some sort of click happens, but it doesn't ensure that action chains click happens. 

Please add more checking that keyword calls the correct Selenium API methods.

Same comment applies also to the test in below.

File: atest/acceptance/keywords/click_element.robot
Comment (aaltat) [inline atest/acceptance/keywords/click_element.robot]: Remove the documentation saying what the test does, it doesn't bring value and only adds maintenance burden. If you want, test name could say: `Click Element With Action Chains`

File: atest/acceptance/keywords/click_element_modifier.robot
Comment (aaltat) [inline atest/acceptance/keywords/click_element_modifier.robot]: Please remove the documentation saying what the test does, it doesn't bring value and only adds maintenance burden. If you want, test name could say: `In Click Element Modified Argument Must 
Overrule Action Chains argument`

File: src/SeleniumLibrary/keywords/element.py
Comment (aaltat) [inline src/SeleniumLibrary/keywords/element.py:601]: Add line here saying: ``action_chain`` is new in SeleniumLibrary 4.1

**Review by aaltat — CHANGES_REQUESTED** [review]
Better but needs few minor enhancements.

File: atest/acceptance/keywords/click_element.robot
Comment (JonKoser) [reply atest/acceptance/keywords/click_element.robot]: True. Unfortunately, ActionChains click() and WebElement.click() register the same JavaScript MouseEvent action so I can't tell them apart that way. I'll see what I can do but if you have any ideas, please let me know!

**Comment by JonKoser** [comment]
Pushed up documentation changes. Still trying to figure out how to tell if it was an "action chain click" or a "web element click"

File: atest/acceptance/keywords/click_element.robot
Comment (JonKoser) [reply atest/acceptance/keywords/click_element.robot]: I'm really at a loss for what I can do with this. I've tried a whole bunch of different things with mockito's spy, spy2, patch, etc. and I can't seem to figure out something that'll work. Do you have any guidance here? The trouble comes from the ActionChains object being created and acted upon only in the scope of `_click_with_action_chain()` - it makes it very difficult to say what was called on that object and what wasn't.

**Comment by aaltat** [comment]
No worries I can take a look tomorrow evening.

**Comment by aaltat** [comment]
Sorry, I did have busy day and I can look today. I will try to find time as soon as possible. Thank yo for being patient.

**Review by aaltat — APPROVED** [review]

**Comment by aaltat** [comment]
If Travis shows green after the builds this is ready to be merged.

**Comment by aaltat** [comment]
Looks like Chrome and Travis are not playing together and builds are failing. Need to fix Travis first.

**Comment by aaltat** [comment]
It looks like I broke something with Python 2, need to investigate closely.

**Comment by aaltat** [comment]
Thank you for the PR.

**Comment by JonKoser** [comment]
Thank you for helping out with this!

**Pull request merged**
